Kconfig.zephyr

```
@@ -397,6 +398,11 @@ config BOOTLOADER_MCUBOOT
 	      (or Armv8-M baseline) targets with no built-in vector relocation
 	      mechanisms
 
+	  By default, this option instructs Zephyr to initialize the core
+	  architecture HW registers during boot, when this is supported by
+	  the application. This removes the need by MCUboot to reset itself
+	  the core registers' state.
```

```suggestion
	  the application. This removes the need by MCUboot to reset
	  the core registers' state itself.
```

Prevent confusion about what is reset.